### PR TITLE
Add functional test to verify generated code is up to date (#1369)

### PR DESCRIPTION
**Pull request by cesarbs**
#1369 

Note it actually wasn't 😝 Thankfully it was just whitespace.

### CODE REVIEW

File: test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/GeneratedCodeTests.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            {
+                var currentFrameHeadersGenerated = File.ReadAllText(frameHeadersGeneratedPath);
+                var currentFrameGenerated = File.ReadAllText(frameGeneratedPath);
+
+                CodeGenerator.Program.Run(testFrameHeadersGeneratedPath, testFrameGeneratedPath);
+
+                var testFrameHeadersGenerated = File.ReadAllText(testFrameHeadersGeneratedPath);
+                var testFrameGenerated = File.ReadAllText(testFrameGeneratedPath);
+
+                Assert.Equal(currentFrameHeadersGenerated, testFrameHeadersGenerated);
```
Comment (natemcmaster) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/GeneratedCodeTests.cs]: Suggest `Assert.Equal(string1, string2, ignoreLineEndingDifferences: true)`

File: tools/CodeGenerator/Program.cs [hunk 2]
```diff
@@ -10,36 +10,38 @@ public class Program
...
+            if (!string.Equals(frameFeatureCollectionContent, existingFrameFeatureCollection))
             {
-                Console.WriteLine(text0);
+                File.WriteAllText(frameFeaturesCollectionPath, frameFeatureCollectionContent);
             }
-            return 0;
         }
     }
-}
+}
```
Comment (natemcmaster) [inline tools/CodeGenerator/Program.cs]: Nit: keep trailing newline.

**Comment by cesarbs** [comment]
Updated.

File: test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/GeneratedCodeTests.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            const string frameHeadersGeneratedPath = "../../../../../src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/FrameHeaders.Generated.cs";
+            const string frameGeneratedPath = "../../../../../src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.Generated.cs";
+
+            var testFrameHeadersGeneratedPath = Path.GetTempFileName();
+            var testFrameGeneratedPath = Path.GetTempFileName();
+
+            try
+            {
+                var currentFrameHeadersGenerated = File.ReadAllText(frameHeadersGeneratedPath);
+                var currentFrameGenerated = File.ReadAllText(frameGeneratedPath);
```
Comment (halter73) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/GeneratedCodeTests.cs]: @pranavkm Do we still delete the src directory when running tests on our CI?

Comment (cesarbs) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/GeneratedCodeTests.cs]: I checked with @natemcmaster, we don't.

**Comment by davidfowl** [comment]
@cesarbs This won't merge anymore. You'll have to rebase on top of dev.

**Pull request merged**